python/lsst/daf/butler/core/ddl.py

```diff
@@ -308,6 +309,11 @@ class FieldSpec:
     doc: Optional[str] = None
     """Documentation for this field."""
 
+    def __post_init__(self) -> None:
```

Comment:
@andy-slac I added this to force the value to False. It does not catch someone creating a `FieldSpec` and then later changing it to `fieldSpec.primaryKey=True` (which is what is seemingly going on in the `addDimensionForeignKey` call but with a default of `False` which is confusing me).

Reply:
It'd probably be cleaner to make `FieldSpec` a frozen dataclass (not just for this), but I bet there's at least one more place where we assume it isn't (in addition to the new `__post_init__`).  So a project for another day, unless you're inspired.

Reply:
The code that started us looking at this is directly manipulating FieldSpec so I know that freezing is not going to work immediately. If we are lucky that's the only place.